libs/tree-sitter-wing/grammar.js

```diff
@@ -56,14 +57,18 @@ module.exports = grammar({
         repeat(seq(".", field("fields", $.identifier)))
       ),
 
-    // This is required because of ambiguity with using Json keyword for both instantiation of Json
-    // and Identifier for static methods. Same issue exists for other types like Set, Map, etc.
-    stdlib_identifier: ($) => choice($._json_types, "str", "num", "bool"),
     nested_identifier: ($) =>
       prec(
         PREC.MEMBER,
         seq(
-          field("object", $.expression),
+          field("object", 
+            choice(
+              $.expression, 
+              // This is required because of ambiguity with using Json keyword for both instantiation of Json
+              // and Identifier for static methods.
+              $.json_container_type
```

Comment:
In our ast's `Reference` we recognize 3 types:
1. `Identifier`.
2. `InstanceMember`.
3. `TypeMember` (static members).

But until now our grammar only recognized the first 2. And the type checker converted 2's to 3's in `resolve_reference` (see expr_maybe_type). But now if we are creating a `nested_identifier` node for `json_container_type` (or other types) we can make our _parser.rs_ detect this can create a `TypeMember` reference automatically.
Also we should probably not call this `object` anymore. Maybe `object_or_type`.

Reply:
This is more or a very specific case for Json, because in our grammar we allow `Json (expression)`. I would say our grammar still only really recognizes the first two expect for this very specific case right now.

I considered going directly to TypeMember but I didn't know if the type checker would have trouble with that be done before expected. I'll try it out.

Reply:
I did get the parser working to generate `TypeMember` appropriately now for the Json.xxx case, will update